### PR TITLE
[#1058] Cleanup dependency injection in reducers

### LINKED ISSUE #1058: Cleanup dependency injection in reducers

### PR DESCRIPTION
**Pull request by LukasKorba**
Closes #1058 

- All work here is just cleanup of obsolete dependency injection. #981 introduced a brand new approach of DI for token name and SDK constants. That allowed me to deliver expected values directly in the TCA fashion so all reducers no longer needed those values to be passed via initializers.

Screenshot 2024-02-14 at 14 35 49

This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md).

# Author

- [x] Self-review: Did you review your own code in GitHub's web interface? Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs.
- [ ] Does the code abide by the [Coding Guidelines](../blob/main/docs/CODING_GUIDELINES.md)?
- [x] Automated tests: Did you add appropriate automated tests for any code changes?
- [x] Code coverage: Did you check the code coverage report for the automated tests?  While we are not looking for perfect coverage, the tool can point out potential cases that have been missed.
- [ ] Documentation: Did you update Docs as appropiate? (E.g [README.md](../blob/main/README.md), etc.)
- [x] Run the app: Did you run the app and try the changes? 
- [ ] Did you provide Screenshots of what the App looks like before and after your changes as part of the description of this PR? (only applicable to UI Changes)
- [x] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit.

# Reviewer

- [ ] Checklist review: Did you go through the code with the [Code Review Guidelines](../blob/main/CODE_REVIEW_GUIDELINES.md) checklist?
- [ ] Ad hoc review: Did you perform an ad hoc review?  _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
- [ ] Automated tests: Did you review the automated tests?
- [ ] Manual tests: Did you review the manual tests?_You will find manual testing guidelines under our [manual testing section](../blob/main/docs/testing/manual_testing)_
- [ ] How is Code Coverage affected by this PR? _We encourage you to compare coverage before and after your changes and when possible, leave it in a better place. [Learn More...](../blob/main/docs/testing/local_coverage.md)_
- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), [LICENSE.md](../blob/main/LICENSE.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
- [ ] Run the app: Did you run the app and try the changes? While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data.

### CODE REVIEW

File: secantTests/ImportWalletTests/ImportWalletTests.swift [hunk 1]
```diff
@@ -18,7 +18,7 @@ class ImportWalletTests: XCTestCase {
         let store = TestStore(
             initialState: .initial
         ) {
-            ImportWalletReducer(saplingActivationHeight: 0)
```
Comment (str4d) [inline secantTests/ImportWalletTests/ImportWalletTests.swift:21]: I don't know if these tests were depending on this value of `saplingActivationHeight`; presumably the test dependency injection will now be normalizing these to the actual testnet Sapling activation height.

File: secantTests/SendTests/SendTests.swift [hunk 2]
```diff
@@ -352,6 +352,9 @@ class SendTests: XCTestCase {
     }
 
     func testDifferentNumberFormats_LiveNumberFormatter() throws {
+        let zcashNumberFormatter = NumberFormatter.zcashNumberFormatter
+        zcashNumberFormatter.locale = Locale(identifier: "en_US")
```
Comment (str4d) [inline secantTests/SendTests/SendTests.swift:356]: Why was this change necessary?

File: secantTests/SendTests/SendTests.swift [hunk 3]
```diff
@@ -724,13 +727,16 @@ class SendTests: XCTestCase {
...
-            SendFlowReducer(networkType: .testnet)
+            SendFlowReducer()
         }
         
         XCTAssertEqual(store.state.message, testMessage.data)
     }
     
     func testFeeFormat() throws {
+        let zashiBalanceFormatter = NumberFormatter.zashiBalanceFormatter
+        zashiBalanceFormatter.locale = Locale(identifier: "en_US")
```
Comment (str4d) [inline secantTests/SendTests/SendTests.swift:738]: Ditto.

File: secantTests/SnapshotTests/PrivateDataConsentSnapshotTests/PrivateDataConsentSnapshotTests.swift [hunk 4]
```diff
@@ -15,7 +15,7 @@ class PrivateDataConsentSnapshotTests: XCTestCase {
         let store = Store(
             initialState: .initial
         ) {
-            PrivateDataConsentReducer(networkType: .mainnet)
```
Comment (str4d) [inline secantTests/SnapshotTests/PrivateDataConsentSnapshotTests/PrivateDataConsentSnapshotTests.swift:18]: Does this test harness correctly initialize the dependency injector to inject mainnet? The vast majority of tests were using testnet, so I presume that is what the default setup for the test dependency injector uses.

**Review by str4d — APPROVED** [review]
utACK 9b683651cf2b077e5a72ef10bead22d8c8dbe477 with questions about the tests.

File: secantTests/ImportWalletTests/ImportWalletTests.swift (hunk 1 above)
Comment (LukasKorba) [reply secantTests/ImportWalletTests/ImportWalletTests.swift:21]: `saplingActivationHeight` is newly read directly inside ImportWalletReducer from a dedicated TCA dependency

File: secantTests/SendTests/SendTests.swift (hunk 2 above)
Comment (LukasKorba) [reply secantTests/SendTests/SendTests.swift:356]: Yes, the simulators have different locale settings and tests were failing. Also CIs use different version of xcode and simulators so I needed to setup the tests with hardcoded locale to US (sometime to CZ so I can check comma instead of point as a floating separator).

File: secantTests/SendTests/SendTests.swift (hunk 3 above)
Comment (LukasKorba) [reply secantTests/SendTests/SendTests.swift:738]: Yes, the simulators have different locale settings and tests were failing. Also CIs use different version of xcode and simulators so I needed to setup the tests with hardcoded locale to US (sometime to CZ so I can check comma instead of point as a floating separator).

File: secantTests/SnapshotTests/PrivateDataConsentSnapshotTests/PrivateDataConsentSnapshotTests.swift (hunk 4 above)
Comment (LukasKorba) [reply secantTests/SnapshotTests/PrivateDataConsentSnapshotTests/PrivateDataConsentSnapshotTests.swift:18]: No, all tests are mostly network independent and while dependent it's always testnet. This injection here is just because the view needed the network from reducer. Newly the reducer can access it directly from TCA dependency.

File: secantTests/SnapshotTests/SecurityWarningSnapshotTests/SecurityWarningSnapshotTests.swift [hunk 5]
```diff
@@ -16,7 +16,7 @@ class SecurityWarningSnapshotTests: XCTestCase {
         let store = Store(
             initialState: .initial
         ) {
-            SecurityWarningReducer(zcashNetwork: ZcashNetworkBuilder.network(for: .mainnet))
```
[the remark this answers is not in the document]
Comment (LukasKorba) [reply secantTests/SnapshotTests/SecurityWarningSnapshotTests/SecurityWarningSnapshotTests.swift:19]: No, all tests are mostly network independent and while dependent it's always testnet. This injection here is just because the view needed the network from reducer. Newly the reducer can access it directly from TCA dependency.

File: secantTests/SnapshotTests/SettingsSnapshotTests/SettingsSnapshotTests.swift [hunk 6]
```diff
@@ -16,7 +16,7 @@ class SettingsSnapshotTests: XCTestCase {
         let store = Store(
             initialState: .initial
         ) {
-            SettingsReducer(networkType: .mainnet)
```
[the remark this answers is not in the document]
Comment (LukasKorba) [reply secantTests/SnapshotTests/SettingsSnapshotTests/SettingsSnapshotTests.swift:19]: No, all tests are mostly network independent and while dependent it's always testnet. This injection here is just because the view needed the network from reducer. Newly the reducer can access it directly from TCA dependency.
